roles/server/tasks/run.yml

```diff
@@ -54,6 +54,18 @@
         ansible.builtin.fail:
           msg: "{{ container_log.stdout_lines }}"
 
+    - name: Wait until cluster formed
+      ansible.builtin.uri:
+        url: http://localhost:11222/rest/v2/cluster?action=distribution
+        user: "{{ user }}"
+        password: "{{ pass }}"
+        method: GET
+        return_content: true
+      register: _result
+      until: "_result.status == 200 and (_result.json | length) == (groups['server'] | length)"
+      retries: 30 # 30 * 5 seconds = 2.5 min
```

Comment:
This seems a bit long. The cluster should already be formed at this point due to waiting for the log. If it isn't a valid status, I would say just throw an error and maybe print the log?

Reply:
Do you mean instead of waiting, we log and exit? Or to change the time?

Reply:
The more I thought about it as I was writing it, I would say just exit. I don't think the cluster will form properly if it hasn't after all the servers have fully started (which includes membership changes)

Reply:
Did you hit this during your tests? I was running into this frequently and I felt it could cause some friction. We would need to restart the server instances again to ensure the cluster forms.

Reply:
I have not run into this at all. However, I don't understand why we would want to wait if they didn't form. I thought you had mentioned the cache is not added appropriately. I am only suggesting to error immediately if they do not join, this way we receive an error sooner and the user can act appropriately (which would most likely be killing ISPN servers).

Reply:
Yeah, this prevents the cache from being improperly initialized. I had cases where the cache was created only in some nodes because we added it before the cluster forms. I'll update it to fail the operation and shut down the running containers.

Reply:
Updated, it should shutdown the container and fail with:

```text
TASK [server : Fail because cluster is malformed] ***************************************************************************************************************************************************************************************************
fatal: [ec2-18-230-117-199.sa-east-1.compute.amazonaws.com]: FAILED! => {"changed": false, "msg": "Cluster is not formed: [{\"node_name\":\"ip-172-31-13-206-46773\",\"node_addresses\":[\"172.31.13.206:7800\"],\"memory_available\":17802640,\"memory_used\":28334704}]"}
fatal: [ec2-54-233-229-101.sa-east-1.compute.amazonaws.com]: FAILED! => {"changed": false, "msg": "Cluster is not formed: [{\"node_name\":\"ip-172-31-12-51-63211\",\"node_addresses\":[\"172.31.12.51:7800\"],\"memory_available\":21087040,\"memory_used\":25050304},{\"node_name\":\"ip-172-31-11-14-23150\",\"node_addresses\":[\"172.31.11.14:7800\"],\"memory_available\":7641192,\"memory_used\":38496152}]"}
fatal: [ec2-18-228-156-201.sa-east-1.compute.amazonaws.com]: FAILED! => {"changed": false, "msg": "Cluster is not formed: [{\"node_name\":\"ip-172-31-12-51-63211\",\"node_addresses\":[\"172.31.12.51:7800\"],\"memory_available\":21087040,\"memory_used\":25050304},{\"node_name\":\"ip-172-31-11-14-23150\",\"node_addresses\":[\"172.31.11.14:7800\"],\"memory_available\":7641192,\"memory_used\":38496152}]"}
```